### PR TITLE
done login tests.

### PR DESCRIPTION
**Pull request by n00b4ever**
https://cmput301.atlassian.net/browse/CMPUT-75

***********
cannot get R.string.users_collection
in FirebaseUserGetSet without importing application context

tried
Resources.getSystem().getString(android.R.string.users_collection).
and
Resources.getSystem().getString(R.string.users_collection). <---- this one pops up 'Users' but doesnt work??????

image

### CODE REVIEW

File: app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/LoginActivityTest.java [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+        solo.assertCurrentActivity(r.getString(R.string.wrong_activity), CreateAccountActivity.class);
+
+        solo.sleep(shortWait);
+    }
+
+    /**
+     * Tests login button functionality with no input
+     */
+    @Test
+    public void emptyLoginTest()
```
Comment (huddy987) [inline app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/LoginActivityTest.java:90]: Can we assert that the error message is shown and the red ! icon? This is an important part of this test I think

File: app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/LoginActivityTest.java [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+    @Test
+    public void invalidEmailTest()
+    {
+        solo.enterText(emailText, r.getString(R.string.wrong_email));
+        solo.enterText(passwordText, r.getString(R.string.wrong_pass));
+        solo.clickOnButton(r.getString(R.string.login));
+
+        solo.sleep(longWait);
+
+        Assert.assertNotNull(emailText.getError());
```
Comment (huddy987) [inline app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/LoginActivityTest.java:141]: Does it make sense to assert the exact error here? What if emailText.getError() is returning an invalid error? For example "input can not be empty"?

File: app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/LoginActivityTest.java [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    @Test
+    public void spacesLoginTest()
+    {
+        solo.enterText(emailText, " ");
+        solo.enterText(passwordText, " ");
+        solo.clickOnButton(r.getString(R.string.login));
+
+        solo.sleep(shortWait);
+
+        Assert.assertNotNull(emailText.getError());
```
Comment (huddy987) [inline app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/LoginActivityTest.java:110]: Can we assert the exact error?

File: app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/LoginActivityTest.java [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    @Test
+    public void wrongPasswordTest()
+    {
+        solo.enterText(emailText, r.getString(R.string.test_account1_email));
+        solo.enterText(passwordText, r.getString(R.string.wrong_pass));
+        solo.clickOnButton(r.getString(R.string.login));
+
+        solo.sleep(longWait);
+
+        Assert.assertNotNull(passwordText.getError());
```
Comment (huddy987) [inline app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/LoginActivityTest.java:126]: Same

**Review by huddy987 — CHANGES_REQUESTED** [review]
Overall nice tests I just had a couple comments

File: app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/LoginActivityTest.java (hunk 1 above)
Comment (n00b4ever) [reply app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/LoginActivityTest.java:90]: sec i am doing so

File: app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/LoginActivityTest.java (hunk 4 above)
Comment (n00b4ever) [reply app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/LoginActivityTest.java:131]: uhm, you cant getError().toString but you can check if text is visible

so if two edittexts both have Cannot be empty! i am only checking once if Cannot be empty exists

But if there are two different error messages showing at the same time,
can check if message1 exists
can check if message 2 exists

Comment (n00b4ever) [reply app/src/androidTest/java/com/example/bookwormadventuresdeluxe2/LoginActivityTest.java:131]: updated

**Review by huddy987 — APPROVED** [review]
Nice

**Review by IsaakC — APPROVED** [review]

**Pull request merged**